src/ecs/component/transform_component.rs

```diff
@@ -37,10 +34,7 @@ impl specs::Component for TransformComponent {
 impl TransformComponent {
     pub fn new(x: f32, y: f32, z: f32, rotation: f32) -> TransformComponent {
         TransformComponent {
-            x: x,
-            y: y,
-            z: z,
-            rotation: rotation,
+            pos: Vector4::new(x, y, z, rotation),
```

Comment:
A bit nit-picky: I'd prefer fully typed out names as opposed to shorthand on the pos/vel members.